NorthstarDedicatedTest/serverauthentication.cpp

```diff
@@ -480,6 +483,12 @@ CCommand__TokenizeType CCommand__Tokenize;
 
 char CGameClient__ExecuteStringCommandHook(void* self, uint32_t unknown, const char* pCommandString)
 {
+	// Only log clientcommands if the convar `ns_should_log_all_clientcommands` equals 1
+	if (Cvar_ns_should_log_all_clientcommands->GetBool())
+	{
+		spdlog::info("{} with UID {} sent stringcommand {}", (char*)self + 0x16, (char*)self + 0xF500, pCommandString);
```

Comment:
If the command contains a `\n`, the client can now write their own console lines (we both know who would figure this out and abuse the hell out of it).

Maybe some sanity checks?
```c
std::string commandStr = (char*)self + 0x16;
		
// Clean indenting/newline characters that would mess up console
for (char& c : commandStr) {
	switch (c) {
		case '\n':
		case '\r':
		case '\t':
			c = ' ';
	}
}
```

Reply:
Doesn't seem to be necessary. From testing, commands are already escaped properly.

For example, I put in, `test\nhello\0abc%soh no` and it printed as:

```
[2022-06-25 12:45:09.344] [info] XXXXXXXXXXX (UID: YYYYYYYYY) executed command: "test\nhello\0abc%soh no"
```

Reply:
Hm ok lol, I wonder about other escape characters... 🤔 I'll test and let yk if I find anything.